#main

// normal try-catch-finally
try {
	//
}
catch ex Exception {
	echo ex.getMessage()
}
finally {
	echo 'do finally'
}

// if with catch-finally
if 1 {
	//
}
else {
	//
}
catch ex {
	echo ex.getMessage()
}
finally {}

// for with catch-finally
for v in [0, 1, 2, 3] {
	throw ErrorException('some message') when v == 2
}
catch ex ErrorException {
	echo ex.getMessage()
}
finally {}

// also supported in while
